coalib/output/ConsoleInteraction.py

```diff
@@ -147,16 +147,44 @@ def acquire_actions_and_apply(console_printer,
             action_dict[metadata.name] = action
             metadata_list.append(metadata)
 
+        applied_action = ask_for_action_and_apply(console_printer,
+                                                  section,
+                                                  metadata_list,
+                                                  action_dict,
+                                                  failed_actions,
+                                                  result,
+                                                  file_diff_dict,
+                                                  file_dict)
+
         # User can always choose no action which is guaranteed to succeed
-        if not ask_for_action_and_apply(console_printer,
-                                        section,
-                                        metadata_list,
-                                        action_dict,
-                                        failed_actions,
-                                        result,
-                                        file_diff_dict,
-                                        file_dict):
+        if not applied_action:
             break
+        else:
+            next_actions = applied_action.next_actions
+
+            # FIXME: currently the code doensn't check for
+            # circular dependencies among the next_actions
+            while next_actions:
+                if len(next_actions) == 1:
+                    # Immediately execute the only action
+                    applied_action = directly_apply_action(next_actions[0],
+                                                           console_printer,
+                                                           section,
+                                                           failed_actions,
+                                                           result,
+                                                           file_diff_dict,
+                                                           file_dict)
+                    next_actions = (applied_action.next_actions
+                                    if applied_action else [])
+
+                if len(next_actions) > 1:
+                    result.actions = next_actions
+                    acquire_actions_and_apply(console_printer,
+                                              section,
+                                              file_diff_dict,
+                                              result,
+                                              file_dict)
+                    break
```

Comment:
Or maybe we omit the outer `while` loop. I can't figure any other use out the use of that "while" loop except 
"Asking the users to choose the same actions again after applying an action, until and unless user chooses to "do nothing".